core/Makefile

```diff
@@ -1,34 +1,42 @@
-dev-discovery-1::
-	@go run main.go -env-file=./infra/dev_config/discovery-one.env
-
-dev-content-1::
-	@go run main.go -env-file=./infra/dev_config/content-one.env
-
-dev-content-2::
-	@go run main.go -env-file=./infra/dev_config/content-two.env
+SQL_SRCS := $(shell find db/sql/ -type f -name '*.sql') db/sqlc.yaml
 
-dev-content-3::
-	@go run main.go -env-file=./infra/dev_config/content-three.env
-
-up:
-	make down
-	make gen
+.PHONY: up down deps test
+up: down gen
 	cd infra && docker compose up --build -d
 
 down:
 	cd infra && docker compose down
 
-deps::
+deps:
 	@go install github.com/onsi/ginkgo/v2/ginkgo@v2.19.0
 	@brew install protobuf
 	@go install github.com/sqlc-dev/sqlc/cmd/sqlc@latest
 	@go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
 	@go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
-
-gen::
+	@go mod tidy
+
+gen: $(SQL_SRCS)
+	@which sqlc || ( \
+		echo "ERROR: sqlc command not found." \
+		"Please run 'make deps' to initialize the environment." \
+		&& false \
+	)
 	cd db && sqlc generate
-	go mod tidy
 
-test::
+test:
 	@go test -v ./...
 	@cd test/integration && ginkgo -r
+
+.PHONY: dev-discovery-1 dev-content-1 dev-content-2 dev-content-3
+dev-discovery-1:
+	@go run main.go -env-file=./infra/dev_config/discovery-one.env
+
+dev-content-1:
+	@go run main.go -env-file=./infra/dev_config/content-one.env
+
+dev-content-2:
+	@go run main.go -env-file=./infra/dev_config/content-two.env
+
+dev-content-3:
+	@go run main.go -env-file=./infra/dev_config/content-three.env
+
```

Comment:
I'm keeping these under the assumption that they're useful for testing, otherwise I'd just as soon delete them since we have the docker compose file.

Reply:
they are as they're quicker than rebuilding the container, maybe it's better to use something like air for hotreloading here